packages/sync-actions/src/product-types-actions.js

```diff
@@ -0,0 +1,205 @@
+      value: newEnum,
+    }),
+    [CHANGE_ACTIONS]: (oldEnum, newEnum) => {
+      const oldEnumInNext = next.values.find(
+        nextEnum => nextEnum.key === oldEnum.key
+      )
+      const changeActions = []
+      if (oldEnumInNext) {
+        if (!deepEqual(oldEnum.label, oldEnumInNext.label)) {
+          changeActions.push({
+            attributeName: next.name,
+            action: getChangeEnumLabelActionName(attributeType),
+            newValue: newEnum,
+          })
+        } else {
+          changeActions.push({
+            attributeName: next.name,
+            action: getChangeEnumOrderActionName(attributeType),
+            value: newEnum,
+          })
+        }
+      } else {
+        changeActions.push({
+          attributeName: next.name,
+          action: 'removeEnumValue',
+          value: oldEnum,
+        })
+        changeActions.push({
+          attributeName: next.name,
+          action: getAddActionName(attributeType),
+          value: newEnum,
+        })
+      }
+      return changeActions
+    },
+    [REMOVE_ACTIONS]: deletedEnum => ({
+      attributeName: next.name,
+      action: 'removeEnumValue',
+      value: deletedEnum,
+    }),
+  })
+  const actions = []
+  const removedKeys = []
+  const newEnumValuesOrder = []
+  flatten(handler(attributeDiff, previous, next)).forEach(action => {
+    if (action.action === 'removeEnumValue') removedKeys.push(action.value.key)
+    else if (action.action === getChangeEnumOrderActionName(attributeType)) {
+      newEnumValuesOrder.push(action.value)
+    } else actions.push(action)
+  })
+  return [
+    ...actions,
+    ...(newEnumValuesOrder.length > 0
+      ? [
+          {
+            attributeName: next.name,
+            action: getChangeEnumOrderActionName(attributeType),
+            values: newEnumValuesOrder,
+          },
+        ]
+      : []),
+    ...(removedKeys.length > 0
+      ? [
+          {
+            attributeName: next.name,
+            action: 'removeEnumValues',
+            keys: removedKeys,
+          },
+        ]
+      : []),
+  ]
+}
+
+export function actionsMapAttributes(
+  attributesDiff,
+  previous,
+  next,
+  diffPaths
+) {
+  const actions = []
+  forEach(attributesDiff, (diffValue, diffKey) => {
+    const extractedPairs = extractMatchingPairs(
+      diffPaths,
+      diffKey,
+      previous,
+      next
+    )
+    if (getIsChangedOperation(diffKey)) {
+      if (Array.isArray(diffValue)) {
+        const deltaValue = diffpatcher.getDeltaValue(diffValue)
+        if (deltaValue.name) {
+          actions.push({
+            action: 'addAttributeDefinition',
+            attribute: deltaValue,
+          })
+        }
+      } else if (getIsAnAttributeDefinitionBaseFieldChange(diffValue)) {
+        actions.push(
+          ...buildBaseAttributesActions({
+            actions: attributeDefinitionsActionsList,
+            diff: diffValue,
+            oldObj: extractedPairs.oldObj,
```

Comment:
Shouldn't these be `new` and `previous` in line with what you said below?

Reply:
the name of the parameter in `buildBaseAttributesActions` and what is given from `extractMatchingPairs` is `oldObj`.
it is so far only in this action file that I started introducing `previous`, and `next`. the rest we have to change in the future. that is if we like the new namings better cc @tdeekens @wizzy25